app/src/client/mod.rs

```diff
@@ -133,11 +133,16 @@ fn MainApp(
     FocusRoot::el([
         UICamera.el(),
         ambient_client_shared::player::PlayerRawInputHandler.el(),
-        WindowSized::el([GameClientView {
+        WindowSized::el([ClientView {
             server_addr,
             user_id,
-            on_loaded: cb(move |client| {
-                let mut game_state = client.game_state.lock();
+            // NOTE: client.game_state is **locked** and accesible through game_state.
+            //
+            // This is to prevent another thread from updating using the client after connection but
+            // just before `on_loaded`. This is a very small window of time, but does occasionally
+            // happen, especially when joining a server which is already running and server
+            // immediately.
```

Comment:
```suggestion
            // happen, especially when immediately joining a server which has just been started.
```

Reply:
Good catch, though the opposite of what I meant. 

It occurs then the server has been running for a while so there is no loading going on and its server immediately